v2/internal/testcommon/kube_per_test_context.go

```diff
@@ -411,6 +411,10 @@ func (tc *KubePerTestContext) CreateResourceAndWait(obj client.Object) {
 
 	gen := obj.GetGeneration()
 	tc.CreateResource(obj)
+	// Only wait for ASO objects
+	if _, ok := obj.(genruntime.MetaObject); !ok {
+		return
+	}
```

Comment:
Doesn't this create a race condition where sometimes the non-ASO resource will be created in time for a later step of the test, and sometimes it won't? This seems risky.

Reply:
Most non-ASO resources (== core Kubernetes resources) such as Secret and ConfigMap don't have a notion of "readiness".

These functions previously panic'ed when called with a non-ASO resource, because right below here we call `tc.Eventually(obj).Should(tc.Match.BeProvisioned(gen))` which internally does `conditioner, ok := obj.(conditions.Conditioner)`, but not every Kubernetes resource implements `Conditioner` (it's ASO specific).

Reply:
Since these panic'ed, this change is safe to make (no callers). If there are callers in the future, it does (closer) to the right thing.

_Technically_ speaking, probably we should have a variant of `v2/internal/testcommon/verify.go` that works for resources that do have a Ready condition but don't implement our interface, such as `Pod`.

In practice, we don't actually make any of those resources in our tests at this time, and if we tried to we _would_ have the problem you're describing, but fixing it is a bigger item and I'm not sure that we really have a need to make Pods or Pod-like objects in our tests.

If we ever do, we can fix this. I'll leave a comment here and the other place I changed to that effect.

Reply:
Nice to have protection here in `Wait` functions. 
We've always been using `CreateResource` function for non-ASO resources.

Reply:
> Not sure if that's what we want?

IMO it is - if there's nothing to wait for, then CreateResourceAndWait should == CreateResource. If we wanted to Wait for Pods or other resources (which we could in the future) we could add specific handling for that here as well (as I mentioned in above comment)